cli/tests/test_fix_command.rs

```diff
@@ -0,0 +1,522 @@
+    insta::assert_snapshot!(content, @"CHANGE D");
+}
+
+#[test]
+fn test_fix_both_sides_of_conflict() {
+    let (test_env, repo_path) = init_with_fake_formatter(&["--uppercase"]);
+    std::fs::write(repo_path.join("file"), "content a\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "a"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-"]);
+    std::fs::write(repo_path.join("file"), "content b\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "b"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "a", "b"]);
+
+    // The conflicts are not different from the merged parent, so they would not be
+    // fixed if we didn't fix the parents also.
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["fix", "-s", "a", "-s", "b"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 3 commits of 3 checked.
+    Working copy now at: mzvwutvl b7967885 (conflict) (empty) (no description set)
+    Parent commit      : qpvuntsm 06fe435a a | (no description set)
+    Parent commit      : kkmpptxz ce7ee79e b | (no description set)
+    Added 0 files, modified 1 files, removed 0 files
+    There are unresolved conflicts at these paths:
+    file    2-sided conflict
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "a"]);
+    insta::assert_snapshot!(content, @r###"
+    CONTENT A
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "b"]);
+    insta::assert_snapshot!(content, @r###"
+    CONTENT B
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "@"]);
+    insta::assert_snapshot!(content, @r###"
+    <<<<<<< Conflict 1 of 1
+    %%%%%%% Changes from base to side #1
+    +CONTENT A
+    +++++++ Contents of side #2
+    CONTENT B
+    >>>>>>> Conflict 1 of 1 ends
+    "###);
+}
+
+#[test]
+fn test_fix_resolve_conflict() {
+    // If both sides of the conflict look the same after being fixed, the conflict
+    // will be resolved.
+    let (test_env, repo_path) = init_with_fake_formatter(&["--stdout", "new content"]);
+    std::fs::write(repo_path.join("file"), "content a\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "a"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-"]);
+    std::fs::write(repo_path.join("file"), "content b\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "b"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "a", "b"]);
+
+    // The conflicts are not different from the merged parent, so they would not be
+    // fixed if we didn't fix the parents also.
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["fix", "-s", "a", "-s", "b"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 3 commits of 3 checked.
+    Working copy now at: mzvwutvl 9c3c9a13 (conflict) (no description set)
```

Comment:
I wonder why this is a conflict

Reply:
I spent some time looking into this. The problem is that `MergedTreeBuilder::write_tree()` doesn't try to resolve conflicts. I'll work on a fix for that but that shouldn't block this PR.

Reply:
#3799 hopefully fixes it

Reply:
Yes, the test turns from `(conflcit)` to `(empty)`.